orca-keel/src/main/kotlin/com/netflix/spinnaker/orca/keel/task/ImportDeliveryConfigTask.kt

```diff
@@ -145,9 +149,14 @@ constructor(
     }
   }
 
-  private fun buildError(errorMessage: String): TaskResult {
-    log.error(errorMessage)
-    return TaskResult.builder(ExecutionStatus.TERMINAL).context(mapOf("error" to errorMessage)).build()
+  private fun buildError(error: Any): TaskResult {
+    val normalizedError = if (error is Map<*, *>) {
+      error["error"] ?: error
+    } else {
+      error.toString()
+    }
+    log.error(normalizedError.toString())
+    return TaskResult.builder(ExecutionStatus.TERMINAL).context(mapOf("error" to normalizedError)).build()
```

Comment:
does `deck` actually parse the `error` key in a stage context?
My general recommendation for making sure errors show up in the UI is to throw and let `RunTaskHandler` put the error info into the json in a consistent way that deck understands
(https://github.com/spinnaker/orca/blob/master/orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/RunTaskHandler.kt#L174)

Reply:
Thanks, @marchello2000. The challenge here is precisely that these errors are not in a format that deck would render in a meaningful/helpful way without additional parsing. In fact, I've started a PR to just that (WIP): https://github.com/spinnaker/deck/pull/7959/

Would you stick by your recommendation in this case? For the record, I did discuss this with @erikmunson and, while I understand I could do the parsing in orca and include markdown in the text, it felt like that responsibility should lie in deck.

Reply:
in an ideal world.... i would like us (on orca side) to have a nice API that sets error in a nice way (other than throwing) but alas... that is prob far away (thought maybe we will get there with plugins orca refactor).

So long as you're adding stuff to deck to parse it, i think it's ok. But it does mean that anyone who is consuming this via API will have to handle both cases...